ocw/lib/EC2.py

```diff
@@ -216,6 +218,105 @@ def cleanup_all(self):
         self.cleanup_images()
         self.cleanup_snapshots()
         self.cleanup_volumes()
+        self.cleanup_uploader_vpcs()
+
+    def delete_vpc(self, region, vpcId):
+        if PCWConfig.getBoolean('cleanup/vpc_only_notify', self._namespace):
+            return False
+        try:
+            vpc = self.ec2_resource(region).Vpc(vpcId)
+            for subnet in vpc.subnets.all():
+                if len(list(subnet.instances.all())):
+                    self.log_warn('{} has associated instance(s) so can not be deleted', vpcId)
+                    return False
+            self.log_info('{} has no associated instances. Initializing cleanup of it', vpcId)
+            for gw in vpc.internet_gateways.all():
+                if self.dry_run:
+                    self.log_info('{} will be not deleted due to dry_run mode', gw)
+                else:
+                    self.log_info('Deleting {}', gw)
+                    vpc.detach_internet_gateway(InternetGatewayId=gw.id)
+                    gw.delete()
+            # delete all route table's
```

Comment:
I'm wondering if it would make sense to refactor those individual routines as functions. Like

    def cleanVpcRouteTable(...)
    def cleanVpcEndoiint(...)

Reply:
nice ! thanks for the hint , not sure why I haven't figure out this :)